xbmc/interfaces/legacy/ListItem.cpp

```diff
@@ -643,6 +644,349 @@ namespace XBMCAddon
+
+        return value.asString();
+      };
+
+      auto getStringArrayProperty = [&getStringProperty](const std::string& key,
+                                                         const CVariant& value) {
+        if (!value.isArray())
+          throw WrongTypeException("%s: %s expects an array", __FUNCTION__, key.c_str());
+
+        std::vector<std::string> values;
+        values.reserve(value.size());
+        for (auto val = value.begin_array(); val != value.end_array(); ++val)
+          values.push_back(getStringProperty(key, *val));
+
+        return values;
+      };
+
+      XBMCAddonUtils::GuiLock lock(languageHook, m_offscreen);
+
+      if (StringUtils::CompareNoCase(type, "video") == 0)
+      {
+        auto& videotag = *GetVideoInfoTag();
+        for (auto prop = data.begin_map(); prop != data.end_map(); ++prop)
+        {
+          auto key = prop->first;
+          StringUtils::ToLower(key);
+          const auto value = prop->second;
+
+          if (key == "dbid")
+            videotag.m_iDbId = getIntegerProperty(key, value);
+          else if (key == "year")
+            videotag.SetYear(getIntegerProperty(key, value));
+          else if (key == "episode")
+            videotag.m_iEpisode = getIntegerProperty(key, value);
+          else if (key == "season")
+            videotag.m_iSeason = getIntegerProperty(key, value);
+          else if (key == "sortepisode")
+            videotag.m_iSpecialSortEpisode = getIntegerProperty(key, value);
+          else if (key == "sortseason")
+            videotag.m_iSpecialSortSeason = getIntegerProperty(key, value);
+          else if (key == "sortseason")
+            videotag.SetEpisodeGuide(getStringProperty(key, value));
+          else if (key == "showlink")
+            videotag.SetShowLink(getStringArrayProperty(key, value));
+          else if (key == "top250")
+            videotag.m_iTop250 = getIntegerProperty(key, value);
+          else if (key == "setid")
+            videotag.m_set.id = getIntegerProperty(key, value);
+          else if (key == "tracknumber")
+            videotag.m_iTrack = getIntegerProperty(key, value);
+          else if (key == "count")
+            item->m_iprogramCount = getIntegerProperty(key, value);
+          else if (key == "rating")
+            videotag.SetRating(getFloatProperty(key, value));
+          else if (key == "userrating")
+            videotag.m_iUserRating = getIntegerProperty(key, value);
+          else if (key == "size")
+            item->m_dwSize = getInteger64Property(key, value);
+          else if (key == "watched") // backward compat - do we need it?
+            videotag.SetPlayCount(getIntegerProperty(key, value));
+          else if (key == "playcount")
+            videotag.SetPlayCount(getIntegerProperty(key, value));
+          else if (key == "overlay")
+          {
+            long overlay = getIntegerProperty(key, value);
+            if (overlay >= 0 && overlay <= 8)
```

Comment:
Not having the codebase in front of me, where does the 8 come from?

Reply:
TBH I blindly copied this from `Listitem.setInfo`...